src/zenml/integrations/bentoml/model_deployers/bentoml_model_deployer.py

```diff
@@ -0,0 +1,449 @@
+                BentoML deployment server with the new model instead of
+                creating and starting a new deployment server.
+            timeout: the timeout in seconds to wait for the BentoML server
+                to be provisioned and successfully started or updated. If set
+                to 0, the method will return immediately after the BentoML
+                server is provisioned, without waiting for it to fully start.
+
+        Returns:
+            The ZenML BentoML deployment service object that can be used to
+            interact with the BentoML model http server.
+        """
+        config = cast(BentoMLDeploymentConfig, config)
+        service = None
+
+        # if replace is True, remove all existing services
+        if replace is True:
+            existing_services = self.find_model_server(
+                pipeline_name=config.pipeline_name,
+                pipeline_step_name=config.pipeline_step_name,
+                model_name=config.model_name,
+            )
+
+            for existing_service in existing_services:
+                if service is None:
+                    # keep the most recently created service
+                    service = cast(BentoMLDeploymentService, existing_service)
+                try:
+                    # delete the older services and don't wait for them to
+                    # be deprovisioned
+                    self._clean_up_existing_service(
+                        existing_service=cast(
+                            BentoMLDeploymentService, existing_service
+                        ),
+                        timeout=timeout,
+                        force=True,
+                    )
+                except RuntimeError:
+                    # ignore errors encountered while stopping old services
+                    pass
+        if service:
+            logger.info(
+                f"Updating an existing BentoML deployment service: {service}"
+            )
+
+            # set the root runtime path with the stack component's UUID
+            config.root_runtime_path = self.local_path
+            service.stop(timeout=timeout, force=True)
+            service.update(config)
+            service.start(timeout=timeout)
+        else:
+            # create a new BentoMLDeploymentService instance
+            service = self._create_new_service(timeout, config)
+            logger.info(f"Created a new BentoML deployment service: {service}")
+
+        return cast(BaseService, service)
+
+    def _clean_up_existing_service(
+        self,
+        timeout: int,
+        force: bool,
+        existing_service: BentoMLDeploymentService,
+    ) -> None:
+        # stop the older service
+        existing_service.stop(timeout=timeout, force=force)
+
+        # delete the old configuration file
+        service_directory_path = existing_service.status.runtime_path or ""
+        shutil.rmtree(service_directory_path)
```

Comment:
This seems a little dangerous, what does `rmtree` do if you pass it an empty string? Delete the current working directory?

I think it would be better if we just do:
```python
if existing_service.status.runtime_path:
  shutil.rmtree(existing_service.status.runtime_path)
```